### PR TITLE
HP timer API

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: src/platform/hp_timer/nrf_802154_hp_timer.h [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+uint32_t nrf_802154_hp_timer_sync_task_get(void);
+
+/**
+ * @brief Set unexpected value for time synchronization.
+ *
+ * This value is used to detect synchronization failure.
+ *
+ * @param[in]  unexpected_value  Unexpected synchronization time.
+ */
+void nrf_802154_hp_timer_sync_invalid_value_set(uint32_t unexpected_value);
```
Comment (KaSroka) [inline src/platform/hp_timer/nrf_802154_hp_timer.h]: I would handle this in HP internally. Rename `nrf_802154_hp_timer_sync_task_get` to `nrf_802154_hp_timer_sync_task_prepare` then calculate and store 'invalid value' inside of this function.

File: src/platform/hp_timer/nrf_802154_hp_timer.h [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+ */
+uint32_t nrf_802154_hp_timer_timestamp_task_get(void);
+
+/**
+ * @brief Get timestamp of last event.
+ *
+ * @returns  Timestamp of last event that triggered the @ref nrf_802154_hp_timer_timestamp_task_get
+ *           task.
+ */
+uint32_t nrf_802154_hp_timer_timestamp_get(void);
```
Comment (KaSroka) [inline src/platform/hp_timer/nrf_802154_hp_timer.h]: Change signature to `bool nrf_802154_hp_timer_timestamp_get(uint32_t * timestamp);`

File: src/platform/hp_timer/nrf_802154_hp_timer.h [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+ *
+ * @returns Current timer value [us].
+ */
+uint32_t nrf_802154_hp_timer_current_time_get(void);
+
+/**
+ * @brief Get task used to synchronize this timer with the LP timer.
+ *
+ * This function configures the timer in order to detect if returned task was triggered to return
+ * valid value by the @ref nrf_802154_hp_timer_sync_task_get.
```
Comment (KaSroka) [inline src/platform/hp_timer/nrf_802154_hp_timer.h]: I think it should ref to `nrf_802154_hp_timer_timestamp_get`

Comment (hubertmis) [reply src/platform/hp_timer/nrf_802154_hp_timer.h]: It should refer to the `nrf_802154_hp_timer_sync_time_get`.

**Pull request merged**